cogito/put.go

```diff
@@ -277,14 +278,32 @@ type gitURL struct {
 	Repo  string
 }
 
+// safeParseUrl parse the given URL and return an error if the URL is not well formated.
+// this avoid to eventually leak passwords in case of error
+//
+// From https://github.com/golang/go/issues/53993
+func safeParse(rawURL string) (*url.URL, error) {
+	parsedUrl, err := url.Parse(rawURL)
+	if err != nil {
+		// check if the returned error is actually of type url.Error
+		if uerr := new(url.Error); errors.As(err, &uerr) {
+			// return only the Err and not the URL to not leak potential passwords
```

Comment:
// Return only the error and not the URL to avoid leaking passwords of the form http://user:password@example.com

Reply:
Mhh. Actually this should be in the function comment, here it is enough

```
// Return only the error and not the URL (as url.Parse does).
```